examples/request_raw.rs

```diff
@@ -0,0 +1,191 @@
+//! Simple example that demonstrates how to use raw API Server requests.
+//! Raw requests are the equivalent of `kubectl get --raw`, and enable
+//! users to query for objects or data that is not available through the
+//! clientsets (i.e. Api<K>, whether primitive or derived from a CRD).
+//! The example builds a tool similar to `kubectl top nodes`.
+use k8s_openapi::{api::core::v1::Node, apimachinery::pkg::api::resource::Quantity};
+use kube::{api::ListParams, Api, ResourceExt};
+use serde::Deserialize;
+
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let client = kube::Client::try_default().await?;
+
+    let api: Api<Node> = Api::all(client.clone());
+    let nodes = api.list(&ListParams::default()).await?;
+
+    let node_names = nodes.items.iter().map(|n| n.name_any()).collect();
+    let mut table = Table::new(node_names);
+
+    for node in nodes.items {
+        // Query node stats by issuing a request to the admin endpoint.
+        // See https://kubernetes.io/docs/reference/instrumentation/node-metrics/
+        let url = format!("/api/v1/nodes/{}/proxy/stats/summary", node.name_any());
+        let req = http::Request::get(url).body(Default::default())?;
+
+        // Deserialize JSON response as a JSON value. Alternatively, a type that
+        // implements `Deserialize` can be used.
+        let resp = client.request::<serde_json::Value>(req).await?;
+
+        // Our JSON value is an object so we can treat it like a dictionary.
+        let summary = resp
+            .get("node")
+            .expect("node summary should exist in kubelet's admin endpoint");
+
+        // The base JSON representation includes a lot of metrics, including
+        // container metrics. Use a `NodeMetric` type to deserialize only the
+        // values we care about.
+        let node_metric = serde_json::from_value::<NodeMetric>(summary.to_owned())?;
+
+        // Get the current allocatable values for the node we are looking at and
+        // save in a table we will use to print the results.
+        let alloc = node.status.unwrap_or_default().allocatable.unwrap_or_default();
+        table.push(node_metric, alloc);
+    }
+
+    table.print();
+
+    Ok(())
+}
+
+/// A stat table made up of rows. Holds a value for the longest name to properly
+/// right pad columns.
+#[derive(Debug, Default)]
+struct Table {
+    rows: Vec<(NodeMetric, NodeAlloc)>,
+    max_name_width: usize,
+}
+
+/// Represents a row in the stat table. A metric is associated with a node and
+/// collects information on the CPU and memory usage
+#[derive(Debug, Deserialize)]
+struct NodeMetric {
```

Comment:
slightly less confusing name, the struct contains many metrics

```suggestion
struct NodeMetrics {
```